### DISCUSSION TITLE
Inactive window opacity

### DISCUSSION THREAD

**Discussion by papuSpartan** [yshui/picom#698] [Q&A] [answered]
Is there a way to whitelist a wm class like for alacritty so that it remains opaque while i have another window focused?

**Comment by tryone144** [comment, marked as answer]
You can specify a list of windows that should always be considered focused with [`focus-exclude` rules](https://github.com/yshui/picom/blob/de959f910b646f2f064cad54a40e931e2ccefd2d/picom.sample.conf#L122-L124). These windows will not be affected by the inactive-opacity and inactive-dim settings.

**Reply by Tcholly** [reply]
Is there a way to exlude all except one? or simply to make it so that only one type of window is affected by transparency?

**Reply by latipun7** [reply]
TLDR: if specific windows exist, don't apply `inactive-opacity` option. how?

In my case, it is something reverse of that OP asked.
Is there a way to whitelist window so that other windows didn't have inactive transparency when that whitelisted window has focus?

My use case is, when I spawn `ksnip`, all other windows have transparency which is not what I want while I want other windows remain opaque so I can take screenshot.

**Reply by papuSpartan** [reply]
I marked the answer as correct but to date I still have not gotten opacity to persist while running discord or certain games. I think it has something to do with exclusive mode (or maybe gamemoded?). I'd be interested to know if any of you have the same problem. For reference I use bspwm.

**Reply by cg00001** [reply]
> TLDR: if specific windows exist, don't apply `inactive-opacity` option. how?
> 
> In my case, it is something reverse of that OP asked. Is there a way to whitelist window so that other windows didn't have inactive transparency when that whitelisted window has focus?
> 
> My use case is, when I spawn `ksnip`, all other windows have transparency which is not what I want while I want other windows remain opaque so I can take screenshot.

I have set PrintScreen key as a shortcut to " ksnip -f" and that solves all issues.

**Reply by absolutelynothelix** [reply]
> Is there a way to exlude all except one? or simply to make it so that only one type of window is affected by transparency?

this should make all windows except firefox windows considered focused:
```
focus-exclude = [
  "class_g != 'firefox'"
];
```
